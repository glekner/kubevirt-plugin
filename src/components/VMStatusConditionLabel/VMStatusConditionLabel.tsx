import * as React from 'react';

import { VMStatusCondition } from '@kubevirt-types';
import { Label, LabelGroup, Popover, PopoverPosition } from '@patternfly/react-core';

export const VMStatusConditionLabel: React.FC<VMStatusCondition> = React.memo((condition) => {
  return (
    <Popover
      position={PopoverPosition.top}
      aria-label="Condition Popover"
      bodyContent={() => <div>{condition?.message}</div>}
    >
      <Label
        color="grey"
        isTruncated
        href="#"
        onClick={(e) => {
          e.preventDefault();
        }}
      >
        {condition?.reason}={condition?.status}
      </Label>
    </Popover>
  );
});
VMStatusConditionLabel.displayName = 'VMStatusConditionLabel';

export const VMStatusConditionLabelList: React.FC<{ conditions: VMStatusCondition[] }> = React.memo(
  ({ conditions }) => {
    return (
      <LabelGroup>
        {conditions.map((condition) => (
          <VMStatusConditionLabel
            key={condition.lastTransitionTime}
            {...condition}
          />
        ))}
      </LabelGroup>
    );
  },
);
VMStatusConditionLabelList.displayName = 'VMStatusConditionLabelList';
