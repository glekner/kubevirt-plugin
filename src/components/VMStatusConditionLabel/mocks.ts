import { VMStatusCondition } from '@kubevirt-types/vm';

export const conditionsMock: VMStatusCondition[] = [
  {
    lastTransitionTime: new Date().toString(),
    message: 'failure',
    reason: 'no_vmi',
    status: 'true',
    type: 'Failure',
  },
  {
    lastTransitionTime: new Date().toString(),
    message: 'success',
    reason: 'ready',
    status: 'true',
    type: 'Ready',
  },
];
