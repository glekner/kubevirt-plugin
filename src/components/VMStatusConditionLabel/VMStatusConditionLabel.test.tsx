import * as React from 'react';

import { cleanup, fireEvent, render, screen } from '@testing-library/react';

import { conditionsMock } from './mocks';
import { VMStatusConditionLabel } from './VMStatusConditionLabel';

afterEach(cleanup);

test('VMStatusConditionLabel', async () => {
  const { reason, status, message } = conditionsMock[0];
  const { asFragment, getByText } = render(<VMStatusConditionLabel {...conditionsMock[0]} />);
  const firstRender = asFragment();

  expect(firstRender).toMatchSnapshot();

  // click on condition to open popover
  fireEvent.click(getByText(`${reason}=${status}`));
  const popoverMessage = await screen.findByText(message);

  expect(popoverMessage).toHaveTextContent(message);
});
