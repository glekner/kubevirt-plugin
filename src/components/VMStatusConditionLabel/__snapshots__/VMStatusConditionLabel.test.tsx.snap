// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render VMStatusConditionLabel 1`] = `
<span
  className="pf-c-label"
  onClick={[Function]}
>
  <a
    className="pf-c-label__content"
    href="#"
  >
    <span
      className="pf-c-label__text"
    >
      no_vmi
      =
      true
    </span>
  </a>
</span>
`;
